wagtail/wagtailadmin/templates/wagtailadmin/pages/listing/_page_title_choose.html

```diff
@@ -6,7 +6,9 @@
 
 <h2>
     {% if page.can_choose %}
-        <a class="choose-page" href="#{{ page.id }}" data-id="{{ page.id }}" data-title="{{ page.get_admin_display_title }}" data-url="{{ page.url }}" data-parent-id="{{ page.get_parent.id }}" data-edit-url="{% url 'wagtailadmin_pages:edit' page.id %}">{{ page.get_admin_display_title }}</a>
+        {% with page=page.specific %}
```

Comment:
It'll probably be more efficient to apply `specific()` to the page queryset in the view (i.e. use `pages = parent_page.get_children().specific()` [here](https://github.com/wagtail/wagtail/blob/21f1292cf75c6c51436ee01d2f3f656c6d7667bf/wagtail/wagtailadmin/views/chooser.py#L77)), as that will only perform one extra database query for each different page type in the list, rather than one for every page.